tests/microos/install_image.pm

```diff
@@ -0,0 +1,96 @@
+    my $image = get_required_var('HDD_1');
+    # Use sda as target disk by default
+    # If we want use nvme disks in the future, we will have to extend bellow logic with custom variable and new partition naming shceme
+    my $device = "/dev/sda";
+    # SL Micro x86_64 image has three partitions, aarch64 and ppc64le images have only two partitions
+    my $root_partition_id = is_x86_64 ? 3 : 2;
+    my $root_partition = "${device}" . $root_partition_id;
+    # New partition id for ignition will be directly after root partition
+    my $ignition_partition_id = $root_partition_id + 1;
+    my $ignition_partition = "${device}" . $ignition_partition_id;
+    record_info("Device information", "Device: ${device}\nRoot partition: ${root_partition}\nIgnition partition: ${ignition_partition}");
+
+    # Mount nfs share with images
+    assert_script_run("mount -o ro,noauto,nofail,nolock -t nfs openqa.suse.de:/var/lib/openqa/share /mnt");
+    assert_script_run("ls -als /mnt/factory/hdd/${image}");
+    # dd image to disk
+    assert_script_run("xzcat /mnt/factory/hdd/${image} | dd of=${device} bs=4096 status=progress");
+    assert_script_run("sync");
+    assert_script_run("umount /mnt");
+
+    my $device_layout = script_output("lsblk");
+    record_info("Device layout", ${device_layout});
+
+    # Modify disk to be able to correctly boot and login
+    assert_script_run("mount ${root_partition} /mnt");
+    assert_script_run("btrfs property set /mnt ro false");
+    # Set correct serial console to be able to see login in first boot
+    assert_script_run("sed -i 's/console=ttyS0,115200/console=ttyS1,115200/g' /mnt/boot/grub2/grub.cfg") if is_x86_64;
+    # Upload original grub configuration
+    upload_logs("/mnt/etc/default/grub", failok => 1);
+    # Set permanent grub configuration
+    assert_script_run("sed -i 's/console=ttyS0,115200/console=ttyS1,115200/g' /mnt/etc/default/grub") if is_x86_64;
+    # Fully disable graphical terminal on legacy systems without UEFI
+    assert_script_run("sed -i 's/#GRUB_TERMINAL=console/GRUB_TERMINAL=console/g' /mnt/etc/default/grub") if (is_ipmi && !get_var('IPXE_UEFI'));
+    # We need to properly set grub terminal bsc#1230844, otherwise grub menu will not be visible in non graphical boot environments
+    assert_script_run("sed -i 's/GRUB_TERMINAL_INPUT=\".*\"/GRUB_TERMINAL_INPUT=\"console gfxterm\"/g' /mnt/etc/default/grub");
+    assert_script_run("sed -i 's/GRUB_TERMINAL_OUTPUT=\".*\"/GRUB_TERMINAL_OUTPUT=\"console gfxterm\"/g' /mnt/etc/default/grub");
+    # Enable root loging with password
+    assert_script_run("echo 'PermitRootLogin yes' > /mnt/etc/ssh/sshd_config.d/root.conf");
+    assert_script_run("btrfs property set /mnt ro true");
+    assert_script_run("umount /mnt");
+
+    # Setup ignition parition on the end of the same disk and resize root partition to use all the space
+    assert_script_run("printf \"fix\n\" | parted ---pretend-input-tty ${device} print");
```

Comment:
I would avoid using assert_script_run with `\n`, it will just time out https://openqa.suse.de/tests/15526622#step/install_image/61
I use script_output despite it designed for something else, it uses script by default which is great for escape characters https://github.com/os-autoinst/os-autoinst/blob/master/distribution.pm#L250

Reply:
@dzedro do we have a ticket for that bug in isotovideo?

Reply:
Either use what @dzedro suggests or consider

```suggestion
    enter_cmd "printf \"fix\n\" | parted ---pretend-input-tty ${device} print; echo PARTED-\$?- > /dev/$serialdev";
    wait_serial 'PARTED-$?';
```

Probably you can just use

```suggestion
    enter_cmd "printf \"fix\n\" | parted ---pretend-input-tty ${device} print";
```

as the next call would ensure synchronicity anyway.

But you might want to consider to replace the whole block of repeated `assert_script_run` calls with its own shell script that is called as a whole

Reply:
> @dzedro do we have a ticket for that bug in isotovideo?

There is no ticket AFAIK

Reply:
I changed it to `script_output`,  it works the same way as before.